File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/category/CategoryResponse.java
package fpt.aptech.server.dto.category;

import lombok.Builder;

/**
 * DTO trả về thông tin danh mục cho Client.
 * Sử dụng Java Record (Java 14+) để tạo object bất biến, hiệu năng cao.
 */
@Builder
public record CategoryResponse(
    Integer id,
    String ctgName,
    Boolean ctgType,
    String ctgIconUrl,
    Integer parentId
) {}

File: APP MOBILE/smart_money/lib/modules/category/models/category_response.dart
/// Response hiển thị thông tin danh mục.
/// Tương ứng: CategoryResponse.java (server)
class CategoryResponse {
  final int id;
  final String ctgName;
  final bool? ctgType;
  final String? ctgIconUrl;
  final int? parentId;
  final String? parentName; // Tên danh mục cha — server trả về để tránh gọi API thừa

  const CategoryResponse({
    required this.id,
    required this.ctgName,
    this.ctgType,
    this.ctgIconUrl,
    this.parentId,
    this.parentName,
  });

  factory CategoryResponse.fromJson(Map<String, dynamic> json) {
    return CategoryResponse(
      id: json['id'] as int,
      ctgName: json['ctgName'] as String,
      ctgType: json['ctgType'] as bool?,
      ctgIconUrl: json['ctgIconUrl'] as String?,
      parentId: json['parentId'] as int?,
      parentName: json['parentName'] as String?,
    );
  }
}



